projects/packages/forms/src/class-file-handler.php

```diff
@@ -0,0 +1,418 @@
+			'timestamp' => time(),
+		);
+
+		error_log( 'DEBUG: Returning successful result: ' . print_r( $result, true ) );
+		return $result;
+	}
+
+	/**
+	 * Filter function to modify the upload directory for forms file uploads.
+	 *
+	 * @param array $upload_dir Original upload directory data.
+	 * @return array Modified upload directory data.
+	 */
+	public function modify_upload_dir( $upload_dir ) {
+		error_log( 'DEBUG: modify_upload_dir called with: ' . print_r( $upload_dir, true ) );
+
+		$forms_base_dir = '/jetpack-forms';
+
+		if ( empty( $upload_dir['subdir'] ) ) {
+			$upload_dir['path']   = path_join( $upload_dir['basedir'], ltrim( $forms_base_dir, '/' ) );
+			$upload_dir['url']    = trailingslashit( $upload_dir['baseurl'] ) . ltrim( $forms_base_dir, '/' );
+			$upload_dir['subdir'] = $forms_base_dir;
+		} else {
+			$new_subdir = $forms_base_dir . $upload_dir['subdir'];
+
+			$upload_dir['path']   = str_replace( $upload_dir['subdir'], $new_subdir, $upload_dir['path'] );
+			$upload_dir['url']    = str_replace( $upload_dir['subdir'], $new_subdir, $upload_dir['url'] );
+			$upload_dir['subdir'] = str_replace( $upload_dir['subdir'], $new_subdir, $upload_dir['subdir'] );
+		}
+
+		$upload_dir['basedir'] = $upload_dir['path'];
+		$upload_dir['baseurl'] = $upload_dir['url'];
+		$upload_dir['error']   = false;
+
+		error_log( 'DEBUG: Returning modified upload_dir: ' . print_r( $upload_dir, true ) );
+		return $upload_dir;
+	}
+
+	/**
+	 * Process file attachments when a form is submitted.
+	 *
+	 * @param int   $post_id   The feedback post ID.
+	 * @param array $all_values All form values.
+	 * @return array The processed file attachments.
+	 */
+	public function process_file_attachments( $post_id, $all_values ) {
+		if ( empty( $all_values ) || ! is_array( $all_values ) ) {
+			return array();
+		}
+
+		$file_attachments = array();
+
+		foreach ( $all_values as $key => $value ) {
+			// Check if this is a file field
+			if ( is_array( $value ) && isset( $value['file_token'] ) ) {
+				$file_data = $this->save_file_from_token( $value['file_token'], $value['name'] );
+				if ( ! is_wp_error( $file_data ) ) {
+					// Store the complete file data directly in the form JSON
+					$file_attachments[ $key ] = $file_data;
+				}
+			}
+		}
+
+		return $file_attachments;
+	}
+
+	/**
+	 * Get the URL for a file based on its ID.
+	 *
+	 * @param string $file_id  The file ID.
+	 * @param int    $post_id  The feedback post ID.
+	 * @return string|false The file URL or false if not found.
+	 */
+	public function get_file_url( $field_id, $file_id, $post_id ) {
+		// Use the WordPress REST API to access the file
+		$rest_url = rest_url( 'wp/v2/feedback/files' );
+		$url      = add_query_arg(
+			array(
+				'field_id' => $field_id,
+				'file_id'  => $file_id,
+				'post_id'  => $post_id,
+			),
+			$rest_url
+		);
+
+		// Add the WordPress REST API nonce
+		$url = add_query_arg( '_wpnonce', wp_create_nonce( 'wp_rest' ), $url );
```

Comment:
I don't think we need the nonce anymore since we are using credentials, or use a jwt